### DISCUSSION TITLE
Beginning needs help - installing 'web-only install'

### DISCUSSION THREAD

**Discussion by ebeyonder** [aaronwmorris/indi-allsky#1663] [Q&A]
I am trying to install the 'web-only install' of indi-allsky found here: https://github.com/aaronwmorris/indi-allsky/wiki/Getting-Started#web-only-install

My aim is to ultimately access indi-allsky over the Internet, similar to how I use Thomas' allsky's web interface (though I understand indi-allsky does not have a similar web interface). I am a beginner at this so please bear with me. I would greatly appreciate any help.

I got myself a 2.5 GB KVM VPS from RackNerd. I proceeded to execute sudo apt-get install git but the 'apt-get' command did not work. I did some Googling and found that the 'yum' command could be used install. I did that and it seems to have worked.

I then cloned the indi-allsky git repository, and navigated to the indi-allsku sub-dir. This is where I encountered a problem when I tried to run setup.sh to install the indi-allsky system.

I got this error message:
Please do not run web_only_setup.sh as root
Re-run this script as the user which will execute the indi-allsky software

I did some Googling and set up a new username - adduser [username]

Then I typed sudo su YOURUSERNAME and got into my new username. I tried running the setup.sh script again but it asked for a password. But I never got the chance to set a password when I used 'adduser'. 

Is there a way around this please? I know this sounds basic but I spent an hour Googling, deleting the username and adding a new one, etc, but could not work around this.

Looking ahead, I see the next step is: Login to the indi-allsky web application https://hostname/

What does this mean really? Do I do this on the SSH app (I am using Terminal on a Mac) or if I did this on an internet computer, what is the 'hostname'? 

I may have more problems when it comes to the SyncAPI part of the install, but perhaps I should get past this first. Grateful for any help. Thank you so much.

**Comment by aaronwmorris** [comment]
The user you create needs to have permissions to become root through the sudo facility.

As root, you can setup the user with sudo access without a password.

```
visudo
```

```
# add the following string to the end of the file
# replace "username" with the username
username ALL=(ALL) NOPASSWD:ALL
```

**Comment by ebeyonder** [comment]
Thanks for helping me, Aaron. I am sorry to trouble you again - I have spent an hour reading up so as not to disturb you but I am really lost.

So after logging in as 'root', I type visudo. I get many lines of words and I cannot enter any further commands. I can scroll up and down  Terminal in MacOS but I cannot edit anything. I understand I am suppose to edit the sudoers file. But I can't even access it. 

I am using the RackNerd KVM and I am wondering if the commands operate differently? 

Thank you for any help.

**Reply by aaronwmorris** [reply]
I was hoping it would not dump you into the VIM editor, but it might have.

See if you can update your editor to nano using this command:

```
sudo update-alternatives --config editor
```

**Comment by ebeyonder** [comment]
Thank you. I did see that option online and tried it - and just tried it again. All I see is this:
[root@racknerd-fb0be58 ~]# sudo update-alternatives --config editor
[root@racknerd-fb0be58 ~]# 

Nothing appears to have happened.

**Reply by ebeyonder** [reply]
I even tried doing this on PuTTy on a windows machine - same outcome.

**Reply by aaronwmorris** [reply]
Maybe try installing nano manually.

```
sudo apt-get install nano
```

**Reply by ebeyonder** [reply]
Thanks. Installed Nano, tried the update-alternatives command and nothing:

Total                                                                                                                      1.6 MB/s | 580 kB     00:00     
Running transaction check
Transaction check succeeded.
Running transaction test
Transaction test succeeded.
Running transaction
  Preparing        :                                                                                                                                   1/1 
  Upgrading        : nano-2.9.8-3.el8_10.x86_64                                                                                                        1/2 
  Running scriptlet: nano-2.9.8-3.el8_10.x86_64                                                                                                        1/2 
  Running scriptlet: nano-2.9.8-1.el8.x86_64                                                                                                           2/2 
  Cleanup          : nano-2.9.8-1.el8.x86_64                                                                                                           2/2 
  Running scriptlet: nano-2.9.8-1.el8.x86_64                                                                                                           2/2 
  Verifying        : nano-2.9.8-3.el8_10.x86_64                                                                                                        1/2 
  Verifying        : nano-2.9.8-1.el8.x86_64                                                                                                           2/2 

Upgraded:
  nano-2.9.8-3.el8_10.x86_64                                                                                                                               

Complete!
[root@racknerd-fb0be58 ~]# sudo update-alternatives --config editor
[root@racknerd-fb0be58 ~]#

**Reply by aaronwmorris** [reply]
Okay, we can just just manually configure sudo.  This is not recommended, but it should work.

```
nano /etc/sudoers.d/username.conf
```

Set the contents like I mentioned above.  Once you are finished, run

```
chown root:root /etc/sudoers.d/username.conf
chmod 440 /etc/sudoers.d/username.conf
```

**Reply by ebeyonder** [reply]
Thanks for continuing to help me with this. Just so as to be clear, when you say set the contents as above, do you mean:


# add the following string to the end of the file
# replace "username" with the username
username ALL=(ALL) NOPASSWD:ALL

If so, can I ask what is it I am to do exactly?

The first line - add the following string - what is the string?

As for the second line, do I find this line somewhere and edit the username?

**Reply by aaronwmorris** [reply]
You can just copy this string `username ALL=(ALL) NOPASSWD:ALL` to a text editor and set the username properly.

**Comment by ebeyonder** [comment]
Thank you so much again, Aaron, for sticking with me. I typed in "nano /etc/sudoers.d/username.conf"

and I got an empty screen with no text at all. I could not type anything in terminal.

If this proves too hard, I may ask someone who knows this to help me out. But it seems that typing in the commands you recommended isn't working. Thank you.

**Reply by aaronwmorris** [reply]
I can create a command for you to run, I just need to know what the username is for your system.

**Comment by ebeyonder** [comment]
Thanks so much Aaron. Do you mean the username I created for this apart from the root account?

**Reply by aaronwmorris** [reply]
yes

**Reply by ebeyonder** [reply]
Thanks so much Aaron. I have sent you an email.

**Comment by ebeyonder** [comment]
I managed to install the web server. However, when I tried to create the API and then input the API into the local server for SyncAPI, I get an error Sync Error 500, transfer failed. I would be grateful for any help.

**Reply by aaronwmorris** [reply]
I would check the webapp logs on the remote side to see what exceptions are being generated.  `/var/log/indi-allsky/webapp-indi-allsky.log`

**Comment by ebeyonder** [comment]
Hi Aaron thanks. The log file is empty. Will it only record errors if there
are errors? Or is the fact that it is empty means that there is a prob?
Thank you.

**Reply by aaronwmorris** [reply]
That probably means the requests are not being received by the remote side.  There may be something wrong with the remote URL in the config.

**Comment by ebeyonder** [comment]
Thank you Aaron. Should the log file on the remote site record anything
else at all?

**Reply by aaronwmorris** [reply]
The log on the remote side should be minimal, but it would only log activity when it receives requests.  Can you post your Support Info dump from your local instance?

**Comment by ebeyonder** [comment]
Thanks Aaron. Do you mean this?

**Reply by aaronwmorris** [reply]
Unfortunately, attachments are not kept from email replies.

**Comment by ebeyonder** [comment]
Trying again please see above.

**Reply by aaronwmorris** [reply]
Looks like you got a 400 error, which is better than a 500 error.  I would really need to see the exception or error that is being generated on the other side.

**Reply by ebeyonder** [reply]
Thanks Aaron. Unfortunately the remote side log file is not showing anything. Any guidance on what I should do to show you that would be appreciated. I am.also happy to grant you full access to the remote server if that helps? Really grateful.

**Reply by aaronwmorris** [reply]
Maybe try rebooting the remote instance.  Sometimes the logs stop working with rsyslog.

**Comment by ebeyonder** [comment]
Here is the log file from the remote side 
![02a95687-799c-4566-835c-f2e3f868edba](https://github.com/user-attachments/assets/28a9b995-b807-47d9-a294-3ded65b24e67)

**Reply by aaronwmorris** [reply]
Can you restart indi-allsky on the local side and see if any new messages are posted on the remote side.  It looks like the API key is properly configured and the times are in sync.

When indi-allsky is first started on the local side, it will create the camera on the remote side via REST call.  I would like to see if there is an error related to creating the remote camera.

**Comment by ebeyonder** [comment]
Hi Aaron I managed to get the server running and for several days it was
mirroring my local PC.

However, recently the website came down and I get this message:

Job for gunicorn-indi-allsky.service failed because of unavailable
resources or another system error.

See "systemctl --user status gunicorn-indi-allsky.service" and "journalctl
--user -xe" for details.

***@***.*** root]$


I suppose this is a server issue?

**Reply by aaronwmorris** [reply]
Can you post your Support Info bundle?  `./misc/support_info.sh`

**Comment by ebeyonder** [comment]
Thanks Aaron. I have sent it to your email.

The strange thing is that every night for maybe 4 days now, the local
system encounters a sync error to the remote system. I have to restart the
remote system and then the local system to get the sync back up again. Then
it works for the day until the same problem repeats overnight. Is this a
racknerd problem?

**Reply by aaronwmorris** [reply]
I would check the log on the remote system `/var/log/indi-allsky/webapp-indi-allsky.log`

The most likely cause of sync failures that I can think of would be time drift on either the local or remote systems.  Part of the authentication of the sync calls requires the time to be relatively correct on both the remote and local systems.  The log might have the cause of the problem.

**Comment by ebeyonder** [comment]
I got an error from your inbox that it is full. Should I email to another
address? The system info is quite long so I don't know if I should just
post it here.

**Reply by aaronwmorris** [reply]
Sorry, I have to periodically delete old emails.  It should be good now if you want to send again.

**Comment by ebeyonder** [comment]
Thanks Aaron. Resent.

**Comment by ebeyonder** [comment]
Hi Aaron it seems that the server is out of space. That's really little
space since I have only 5 days worth of media. How do you manage the space
issue on Racknerd? Thank you.

**Comment by aaronwmorris** [comment]
The images can take up a lot of space.  You can setup a more aggressive expiration of the images on the remote system, such as 2 days using the `./misc/expire_images.py`

On my Racknerd instance, I actually do not store any images or video, I upload all of those to S3 and serve them from there.  There is more cost by doing it this way, but it is not too bad.
